File: Algorithms/905.sort-array-by-parity/905.sort-array-by-parity.go
func sortArrayByParity(A []int) []int {
	len := A.size - 1
	for i := 0;i < len;i++ {
		if A[i] % 2 == 1 {
			// A[i--] = temp
			// A[len--] = A[i--]
			A[i--],A[len--] = A[len--],A[i--]

		}
	}
	retuen A
}

File: Algorithms/905.sort-array-by-parity/905.sort-array-by-parity.py
class Solution(object):
    def sortArrayByParity(self, A):
        """
        :type A: List[int]
        :rtype: List[int]
        """
        i, j = 0, len(A) - 1
        while i < j:
            if A[i] % 2 > A[j] % 2:
                A[i], A[j] = A[j], A[i] #互换

            if A[i] % 2 == 0: i += 1
            if A[j] % 2 == 1: j -= 1

        return A
        
